res/controllers/Numark-Mixtrack-Platinum-scripts.js

```diff
@@ -0,0 +1,1415 @@
+        key: 'bpm_tap',
+        off: 0x01,
+        connect: function () {
+            this.group = '[Channel' + eu.currentUnitNumber + ']';
+            components.Button.prototype.connect.call(this);
+        },
+        input: function (channel, control, value, status, group) {
+            components.Button.prototype.input.call(this, channel, control, value, status, group);
+            if (this.isPress(channel, control, value, status)) {
+                eu.forEachComponent(function (component) {
+                    if (component.tap !== undefined) {
+                        component.tap();
+                    }
+                });
+            }
+            else {
+                eu.forEachComponent(function (component) {
+                    if (component.untap !== undefined) {
+                        component.untap();
+                    }
+                });
+            }
+        },
+    });
+
+    this.EffectEnableButton = function (number) {
+        this.number = number;
+        this.group = '[EffectRack1_EffectUnit' + eu.currentUnitNumber +
+                      '_Effect' + this.number + ']';
+        this.midi = [0x97 + eu.currentUnitNumber, this.number - 1];
+        this.inToggle = components.Button.prototype.inToggle;
+        this.flash_timer = null;
+
+        components.Button.call(this);
+    };
+    this.EffectEnableButton.prototype = new components.Button({
+        type: components.Button.prototype.types.powerWindow,
+        outKey: 'enabled',
+        inKey: 'enabled',
+        off: 0x01,
+        tap: function() {
+            this.inKey = 'enabled';
+            this.type = components.Button.prototype.types.toggle;
+            this.inToggle = this.toggle_focused_effect;
+        },
+        untap: function() {
+            this.type = components.Button.prototype.types.powerWindow;
+            this.inToggle = components.Button.prototype.inToggle;
+        },
+        shift:  function() {
+            this.inKey = 'next_effect';
+            this.type = components.Button.prototype.types.push;
+        },
+        unshift: function() {
+            this.inKey = 'enabled';
+            this.type = components.Button.prototype.types.powerWindow;
+        },
+        output: function(value, group, control) {
+            var focused_effect = engine.getValue(eu.group, "focused_effect");
+            if (focused_effect !== this.number) {
+                engine.stopTimer(this.flash_timer);
+                this.flash_timer = null;
+                components.Button.prototype.output.call(this, value, group, control);
+            }
+            else {
+                this.startFlash();
+            }
+        },
+        toggle_focused_effect: function() {
+            if (engine.getValue(eu.group, "focused_effect") === this.number) {
+                engine.setValue(eu.group, "show_focus", 0);
+                engine.setValue(eu.group, "show_parameters", 0);
```

Comment:
I'm not sure this should toggle `show_parameters`. I would presume that mostly users with this controller would be using effect focusing to control the metaknob with the touchstrip, in which case showing all the parameters would be an annoying waste of space. On the other hand, I could imagine that a few users would be using the mixer knobs to adjust the individual parameters when an effect is focused. I'm a bit hesitant to add yet another option for this...

Reply:
I was on the fence about this. When I focus an effect, I use it to change the parameters. Though if you don't do that, I can see this being an issue. The only way around it would be adding an option. Side note: Maybe we need a UI for controller mapping options. That's something I'm willing to work on.

Reply:
Okay, let's make it an option then. I agree, we should have a GUI for controller mapping options. But before making major changes to the mapping system I think we should finish #1795.

Reply:
Updated. I left setting `show_focus` enabled unconditionally, what are your thoughts on that? Should that also fall under the option to show the parameters?

Reply:
Yes, `show_focus` should be toggled regardless of the option. There should not be a case where there is focus but the user can't see it.

Reply:
I believe there is one case, if you exit mixxx with an effect focused, that effect will be focused when you start again (the controller button will flash), but the ui will not show it as focused. I'd have to double check. This is something of a weird edge case though.

Reply:
Hmm, if you can reproduce it try unplugging your controller before starting Mixxx again. There may be something in the mapping causing that.

Reply:
Yup. Tested this again. If you exit with an effect focused, it remains focused when you start again, but the ui doesn't show it. Should I make the mapping check for this on startup and enable `show_focus`?